src/services/signatureHelp.ts

```diff
@@ -525,32 +534,35 @@ function getSpreadElementCount(node: SpreadElement, checker: TypeChecker) {
     return 0;
 }
 
-function getArgumentCount(argumentsList: Node, ignoreTrailingComma: boolean, checker: TypeChecker) {
+function getArgumentCount(argumentsList: Node, checker: TypeChecker) {

```

Comment:
This function and `getArgumentIndex` are meant to be kinda in sync - I can try to merge them if there is an appetite for it.

Reply:
I am somewhat curious how that'd look, but I also don't want you to spend a bunch of time on that if you don't want to.

Reply:
> if you don't want to.

this tbh ;p

IIRC they are now mostly the same, one call would pass an "end index" to get the argument's index and the other one could pass `undefined` in its place to iterate through the whole thing and get the arg count (then we'd also have to fixup the index vs count difference with some `+ 1`)

Reply:
Skimming, aren't these the same functions, just `getArgumentIndex` breaks out early when it finds a specific node? The count difference only matters if you actually finish the loop. It does feel like they should be just one iterator that returns early when a given node is found (or just lets it end if not provided).

Reply:
yep, I mostly ended up not unifying them because I had to switch to other work and because naming things is hard 😉 if you want them unified (which is a reasonable ask) I'll get to it some time later

Reply:
It's pretty easy to unify them:

```ts
function getArgumentIndexOrCount(argumentsList: Node, checker: TypeChecker, node: Node | undefined) {
    const args = argumentsList.getChildren();
    let argumentIndex = 0;
    let skipComma = false;
    for (const child of args) {
        if (node && child === node) {
            if (!skipComma && child.kind === SyntaxKind.CommaToken) {
                argumentIndex++;
            }
            return argumentIndex;
        }
        if (isSpreadElement(child)) {
            argumentIndex += getSpreadElementCount(child, checker);
            skipComma = true;
            continue;
        }
        if (child.kind !== SyntaxKind.CommaToken) {
            argumentIndex++;
            skipComma = true;
            continue;
        }
        if (skipComma) {
            skipComma = false;
            continue;
        }
        argumentIndex++;
    }
    if (node) {
        return argumentIndex;
    }
    return args.length && last(args).kind === SyntaxKind.CommaToken ? argumentIndex + 1 : argumentIndex;
}
```

The harder part is reintegrating the comments, I think.

Reply:
I've used this suggestion and reintegrated the comments.

Reply:
Thanks; I forgot to mention it but would you mind leaving the original functions behind as wrappers, just so the call sites remain the same? (A la abcdWorker funcs in the checker, for example)

Reply:
I'm also totally happy to apply that change; I think this is something I want to suggest for the next patch release just due to the crashes.

Reply:
Pushed that to your branch quick just to move this forward towards a backport (sorry to make you pull)